### ISSUE TITLE
How to set the format of float numbers when `print!()`

### ISSUE THREAD

**Issue by Ionizing** [ExpHP/vasp-poscar#18]
The original POSCAR:
```
ammonia flipping
   1.00000000000000
     6.000000    0.000000    0.000000
     0.000000    7.000000    0.000000
     0.000000    0.000000    8.000000
     H     N
     3     1
Selective
Direct
  0.636428  0.567457  0.5491645   T T T
  0.500000  0.364985  0.5491330   T T T
  0.363572  0.567457  0.5491645   T T T
  0.500000  0.500000  0.5000000   F F F
```

my code:
```rust
use vasp_poscar::{Poscar};

fn main() {
    let poscar = match Poscar::from_path("POSCAR") {
        Ok(p) => p,
        Err(_) => return,
    };
    print!("{}", poscar);
}
```

The result:
```
ammonia flipping
  1.0
    6.0 0.0 0.0
    0.0 7.0 0.0
    0.0 0.0 8.0
   H  N
   3  1
Selective Dynamics
Direct
  0.636428 0.567457 0.5491645 T T T
  0.5 0.364985 0.549133 T T T
  0.363572 0.567457 0.5491645 T T T
  0.5 0.5 0.5 F F F
```

Are there any workarounds that let me format the float numbers of position when displaying them? After all, a formatted POSCAR would be easy to read for human.

**Comment by ExpHP** [comment]
I believe I could make it so that formatting flags are applied to all printed floats.  This way, you could achieve this with
```rust
print!("{:.6}", poscar);

// or maybe
print!("{:-9.6}", poscar);
```
or etc.  It might require a bit of a trick though to allow it to keep using `dtoa` for `{}` (which I want to do because rust's default formatter is horrendous...).

If you need to have different formats for scale factor, lattice, and coords, that'll be a bit more involved.

**Comment by Ionizing** [comment]
Would it be possible that a prettier default format (e.g. `{:10.6}`) is provided rather than the naked style ?

By the way, I tried the `print!("{:.6}", poscar);` but it did not work yet. It seems this feature is still on TODO list ?

**Comment by ExpHP** [comment]
> Would it be possible that a prettier default format (e.g. `{:10.6}`) is provided rather than the naked style ?

Hm.  I'm not sure that I want to compromise on the default behavior.  I want `dtoa` to remain the default for `"{}"`, for the following reasons:

* A core design goal of the crate is to easily allow files to be written and read with no roundoff error (beyond `f64` precision) by default, allowing a printed file to precisely convey the data contained in memory.  This is, for instance, why it does not automatically absorb the scale factor into the lattice; all roundoff in this crate is opt-in.
  This is for (a) reliable communication between programs, and (b) for debugging. For instance, it can be useful to print a structure and see that one of its atoms has fractional `x` coordinate around `-1e-30`, as this is a value that can easily break naively-written software (most implementations of `mod_floor(-1e30, 1.0)` produce `1.0` instead of `0.0`).
* People may disagree on what a good default is for rounding.  For instance, I regularly work with large structures with lattice constants on the order of hundreds or thousands of angstrom, so I might have different ideals for a good number of digits before/after the decimal than some others.
* Users who know what format they want could easily write their own function that is a wrapper around `format!` or `print!`. e.g.
  ```rust
  fn format_poscar(poscar: &Poscar) -> String {
      format!("{:10.6f}", poscar)
  }
  ```


> By the way, I tried the print!("{:.6}", poscar); but it did not work yet. It seems this feature is still on TODO list ?

This is not implemented, I am merely suggesting a way that I could add this feature.

Basically I am asking, "if I added this, would that be good enough for your use case?"  If so I can try adding this feature today.

**Comment by Ionizing** [comment]
> Hm. I'm not sure that I want to compromise on the default behavior. I want dtoa to remain the default for "{}", for the following reasons

I've got your points, thank you very much.

> "if I added this, would that be good enough for your use case?"

In fact, I just finished primary Rust and I am working on a tiny tool to manipulate POSCAR for practice. Though it seems my 'use case' is not so important, I'll appreciate it if that could be added. I'm sure it would be great.
